packages/cypress/src/generators/configuration/configuration.ts

```diff
@@ -149,9 +149,14 @@ In this case you need to provide a devServerTarget,'<projectName>:<targetName>[:
     throw new Error('Either baseUrl or devServerTarget must be provided');
   }
 
+  const nxJson = readNxJson(tree);
+  const addPlugin =
+    process.env.NX_ADD_PLUGINS !== 'false' &&
+    nxJson.useInferencePlugins !== false;
```

Comment:
This generator receives an `addPlugin` option that can override this. We need to respect the option if provided:
```suggestion
  let addPlugin = options.addPlugin;
  if (addPlugin === undefined) {
    const nxJson = readNxJson(tree);
    addPlugin =
      process.env.NX_ADD_PLUGINS !== 'false' &&
      nxJson.useInferencePlugins !== false;
  }
```